module/behaviour/strategy/PS3Walk/src/PS3Walk.cpp

```diff
@@ -17,123 +17,144 @@
                     }
                 }
-                // If it's closed then we should try to reconnect
-                else if (!joystick.valid()) {
-                    joystick.reconnect();
+            });
+
+            on<Trigger<HeadControlEvent>>().then([this](const HeadControlEvent& event) {
+                if (event.pressed) {
+                    if (headLocked) {
+                        log<NUClear::INFO>("Head unlocked");
+                        headLocked = false;
+                    }
+                    else {
+                        log<NUClear::INFO>("Head locked");
+                        headLocked = true;
+                    }
+                }
+            });
+
+            on<Trigger<TriggerActionEvent>>().then([this](const TriggerActionEvent& event) {
+                if (event.pressed) {
+                    log<NUClear::INFO>("Triggering actions");
+                    emit(std::make_unique<MotionCommand>(utility::behaviour::StandStill()));
+                    emit(std::make_unique<ActionPriorites>(ActionPriorites{id, {90}}));
+                    emit(std::make_unique<ExecuteScriptByName>(id, actions));
+                }
+            });
+
+            on<Trigger<ScriptTriggerEvent>>().then([this](const ScriptTriggerEvent& event) {
+                if (event.pressed) {
+                    log<NUClear::INFO>("Standing");
+                    emit(std::make_unique<MotionCommand>(utility::behaviour::StandStill()));
+                    emit(std::make_unique<ActionPriorites>(ActionPriorites{id, {90}}));
+                    emit(std::make_unique<ExecuteScriptByName>(id, "Stand.yaml"));
+                }
+            });
+
+            on<Trigger<LeftKickEvent>>().then([this](const LeftKickEvent& event) {
+                if (event.pressed) {
+                    log<NUClear::INFO>("Requesting Left Front Kick");
+                    // vector pointing forward relative to robot
+                    emit(std::make_unique<KickScriptCommand>(
+                        KickScriptCommand{Eigen::Vector3d(1.0, 0.0, 0.0), LimbID::LEFT_LEG}));
+                }
+            });
+
+            on<Trigger<RightKickEvent>>().then([this](const RightKickEvent& event) {
+                if (event.pressed) {
+                    log<NUClear::INFO>("Requesting Right Front Kick");
+                    // vector pointing forward relative to robot
+                    emit(std::make_unique<KickScriptCommand>(
+                        KickScriptCommand(Eigen::Vector3d(1.0, 0.0, 0.0), LimbID::RIGHT_LEG)));
+                }
+            });
+
+            on<Trigger<WalkVelocityEvent>>().then([this](const WalkVelocityEvent& event) {
+                switch (event.direction.value) {
+                    case WalkVelocityEvent::Direction::HORIZONTAL:
+                        // y is left relative to robot
+                        // strafe[1] = -joystick.value;
+                        walkCommand.z() = -event.value;
```

Comment:
Is there supposed to be some limit or scaling on this?

Reply:
scaling happens when we send the message